### PR TITLE
Prepare TDS_ConfigService for use with the maven-release-plugin

### PR DESCRIPTION
**Pull request by wtritch**
These changes are preparing the TDS_ConfigService for releasing using the maven-release-plugin.

### CODE REVIEW

File: client/pom.xml [hunk 1]
```diff
@@ -5,7 +5,6 @@
     <modelVersion>4.0.0</modelVersion>
 
     <artifactId>tds-config-client</artifactId>
-    <version>0.0.1-SNAPSHOT</version>
```
Comment (wtritch) [inline client/pom.xml]: Changed to inherit the parent POM version.

File: client/pom.xml [hunk 2]
```diff
@@ -48,7 +47,6 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>${maven.compiler.plugin.version}</version>
```
Comment (wtritch) [inline client/pom.xml]: Changed to use dependency-management from the parent POM

File: client/pom.xml [hunk 3]
```diff
@@ -75,4 +72,26 @@
...
+                                <goals>
+                                    <goal>jar</goal>
+                                </goals>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+    </profiles>
```
Comment (wtritch) [inline client/pom.xml]: Added a source jar to the release process.

File: pom.xml [hunk 4]
```diff
@@ -27,14 +34,63 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <docker.image.prefix>fwsbac</docker.image.prefix>
         <spring.cloud.config.version>1.2.2.RELEASE</spring.cloud.config.version>
+        <mockito.version>1.10.19</mockito.version>
+        <joda-time.version>2.9.5</joda-time.version>
 
-        <tds.config.client.version>0.0.1-SNAPSHOT</tds.config.client.version>
         <tds.common.version>0.0.1</tds.common.version>
         <tds-common-legacy.version>0.0.1</tds-common-legacy.version>
-        <tds.student.client.version>0.0.1-SNAPSHOT</tds.student.client.version>
```
Comment (millerts) [inline pom.xml]: Was this unused?

File: pom.xml [hunk 5]
```diff
@@ -9,6 +9,13 @@
     <packaging>pom</packaging>
     <version>0.0.1-SNAPSHOT</version>
 
+    <scm>
+        <connection>scm:git:https://github.com/SmarterApp/TDS_ConfigService.git</connection>
+        <developerConnection>scm:git:https://git@github.com/SmarterApp/TDS_ConfigService.git</developerConnection>
+        <url>https://github.com/SmarterApp/TDS_ConfigService</url>
+        <tag>HEAD</tag>
+    </scm>
```
Comment (wtritch) [inline pom.xml]: Configured maven with the SCM values that allow the maven-release-plugin to push version bump commits.

File: pom.xml [hunk 6]
```diff
@@ -27,14 +34,63 @@
...
 
         <maven.compiler.plugin.version>3.5.1</maven.compiler.plugin.version>
-        <mockito.version>1.10.19</mockito.version>
-        <joda-time.version>2.9.5</joda-time.version>
+        <maven.jar.plugin.version>2.3.2</maven.jar.plugin.version>
+        <maven.release.plugin.version>2.5.3</maven.release.plugin.version>
+        <maven.source.plugin.version>2.1.2</maven.source.plugin.version>
+        <maven.toolchains.plugin.version>1.1</maven.toolchains.plugin.version>
+        <publish.package.name>tds.config</publish.package.name>
     </properties>
```
Comment (wtritch) [inline pom.xml]: Consolidated versions here.

File: pom.xml [hunk 7]
```diff
@@ -27,14 +34,63 @@
...
+        </plugins>
+    </build>
+
+    <distributionManagement>
+        <repository>
+            <id>bintray-smarter-app-open-test-system</id>
+            <name>smarter-app-open-test-system</name>
+            <url>https://api.bintray.com/maven/smarter-app/open-test-system/${publish.package.name}/;publish=1</url>
+        </repository>
+    </distributionManagement>
```
Comment (wtritch) [inline pom.xml]: Created a "tds.config" package in Bintray and configured releases to publish to the package.

File: service/src/main/docker/Dockerfile [hunk 8]
```diff
@@ -9,7 +9,7 @@ FROM frolvlad/alpine-oraclejdk8:slim
 VOLUME /tmp
 
 COPY docker-startup.sh docker-startup.sh
-COPY tds-config-service-0.0.1-SNAPSHOT.jar tds-config-service.jar
+COPY tds-config-service-*.*.*-SNAPSHOT.jar tds-config-service.jar
```
Comment (wtritch) [inline service/src/main/docker/Dockerfile]: Hrm... this will need to be updated to account for non-SNAPSHOT versions.

File: pom.xml (hunk 4 above)
Comment (wtritch) [inline pom.xml]: Yessir, and it's a good thing, too.  I didn't see any circular dependencies in my once-over.

**Pull request merged**